packages/web3-providers-ws/src/index.js

```diff
@@ -101,7 +101,10 @@ WebsocketProvider.prototype.constructor = WebsocketProvider;
  * @returns {void}
  */
 WebsocketProvider.prototype.connect = function () {
-    this.connection = new Ws(this.url, this.protocol, undefined, this.headers, this.requestOptions, this.clientConfig);
+    // TODO: include this.requestOptions and this.clientConfig
```

Comment:
This TODO is important. Relevant Web3 API docs are [here][1] plus recent additions in #3566

[1]: https://web3js.readthedocs.io/en/v1.2.9/web3.html#configuration